tiatoolbox/models/architecture/vanilla.py

```diff
@@ -79,6 +80,48 @@ def _get_architecture(
     return model.features
 
 
+def _get_timm_architecture(
+    arch_name: str,
+    weights: str or WeightsEnum = "DEFAULT",
+    **kwargs: dict,
+) -> list[nn.Sequential, ...] | nn.Sequential:
+    """Get architecture and weights for pathology-specific timm models.
+
+    Args:
+        arch_name (str):
+            Architecture name.
+        weights:
+            path to pretrained weights
+        kwargs (dict):
+            Key-word arguments.
+
+    Returns:
+        A ready-to-use timm model.
+    """
+    if arch_name == "uni_v1":
+        # UNI tile encoder: https://huggingface.co/MahmoodLab/UNI
+        feat_extract = timm.create_model(
+            "hf-hub:MahmoodLab/UNI",
+            pretrained=True,
+            init_values=1e-5,
+            dynamic_img_size=True,
+        )
+    elif arch_name == "prov-gigapath":
+        # ProViT-GigaPath tile encoder: https://huggingface.co/prov-gigapath/prov-gigapath
+        # does not work with timm==0.9.8, needs timm==1.0.3: https://github.com/prov-gigapath/prov-gigapath/issues/2
+        assert (
```

Comment:
This should raise Error https://docs.astral.sh/ruff/rules/assert/

Reply:
It did, should I ignore it with NOQA? Or remove?

Ignoring will allow to avoid unnecessary silent errors while creating the prov-gigapath model.

Reply:
removed in commit [6ec9ca1](https://github.com/TissueImageAnalytics/tiatoolbox/pull/856/commits/6ec9ca14a1ff0f6a443668ce8fe083d3a2dbe9e8)